vaxrank/core_logic.py

```diff
@@ -83,15 +95,18 @@ def vaccine_peptides_for_variant(
             candidate_vaccine_peptide.combined_score)
         candidate_vaccine_peptides.append(candidate_vaccine_peptide)
 
-    max_score = max(vp.combined_score for vp in candidate_vaccine_peptides)
     n_total_candidates = len(candidate_vaccine_peptides)
+    if n_total_candidates == 0:
+        logger.info("No candidate peptides for variant %s", variant.short_description)
+        return candidate_vaccine_peptides
 
+    max_score = max(vp.combined_score for vp in candidate_vaccine_peptides)
     # only keep candidate vaccines that are within 1% of the maximum
     # combined score
     filtered_candidate_vaccine_peptides = [
         vp
         for vp in candidate_vaccine_peptides
-        if vp.combined_score / max_score > 0.99
+        if _isclose(max_score, 0.0) or vp.combined_score / max_score > 0.99
```

Comment:
Maybe do this check outside the list comprehension?

```
if np.isclose(max_score, 0.0):
    filtered_vaccine_peptides = candidate_vaccine_peptides
else:
    filtered_vaccine_peptides = ... # old list comprehension
```

I also don't remember at which point we would filter out vaccine peptides whose score is 0.